### PR TITLE
Adds block storage extend's action support

### PR DESCRIPTION
**Pull request by marioluan**
This aims to fix https://github.com/gophercloud/gophercloud/issues/66
Openstack's source code: https://github.com/openstack/cinder/blob/8.1.0/cinder/volume/drivers/disco/disco.py#L379

### CODE REVIEW

**Comment by jrperritt** [comment]
You'll need to remove your Godeps and .gitignore files

**Comment by marioluan** [comment]
@jrperritt done!

File: openstack/blockstorage/v2/extensions/volumeactions/requests.go [hunk 1]
```diff
@@ -172,3 +172,35 @@ func TerminateConnection(client *gophercloud.ServiceClient, id string, opts Term
...
+// ExtendSizeOptsBuilder allows extensions to add additional parameters to the
+// ExtendSize request.
+type ExtendSizeOptsBuilder interface {
+	ToVolumeExtendSizeMap() (map[string]interface{}, error)
+}
+
+// ExtendSizeOpts contain options for extending the size of an existing Volume. This object is passed
+// to the volumes.ExtendSize function.
+type ExtendSizeOpts struct {
+	NewSize int `json:"new_size,omitempty"`
```
Comment (jrperritt) [inline openstack/blockstorage/v2/extensions/volumeactions/requests.go]: What units is `NewSize`? GBs?

Should this be required? If so:
```
NewSize int `json:"new_size" required:"true"`
```

What happens if a value less than the current size is provided?

Comment (marioluan) [inline openstack/blockstorage/v2/extensions/volumeactions/requests.go]: @jrperritt
I've added the required tag to the field and also added a documentation to it, regarding the format (GB).

Openstack returns an error if the value is less than the current size. Should I do something about it?

Comment (jrperritt) [inline openstack/blockstorage/v2/extensions/volumeactions/requests.go]: >  Should I do something about it?

No, I was just curious if OpenStack would downsize as well. Nothing we can do without separately querying the current size.

**Comment by jrperritt** [comment]
+2

**Pull request merged**